chore: add release blogpost for Kuma 1.6.0 (#773)
kumahq/kuma-website: docs/_blog/2022-04-12-kuma-1-6-0.md

---
title: Kuma 1.6.0 GA Released With Kubernetes Gateway API support, ZoneEgress improvements, a new and improved transparent-proxy and more.
description: Kuma 1.6.0 GA Released With Kubernetes Gateway API support, ZoneEgress improvements, a new and improved transparent-proxy and more.
date: 2022-04-12
headerImage: /images/blog/kuma_1_6_blog_banner.jpg
tags:
  - Release
canonicalUrl: 'https://kuma.io/blog/kuma-1-6-0-released'
---

We are happy to announce Kuma's latest release, which is packed with features and improvements, including substantial performance improvements to our built gateway and .

We strongly suggest upgrading, in order to take advantage of the latest and greatest when it comes to service mesh.

## Notable Features

> * 🚀 We provide a preview of [Kubernetes Gateway API](https://gateway-api.sigs.k8s.io/) support for our builtin gateway. This makes it easier than to provide a gateway to lead traffic through your mesh

Reviewer: ```suggestion
* 🚀 We provide a preview of [Kubernetes Gateway API](https://gateway-api.sigs.k8s.io/) support for our builtin gateway. This makes it easier than to provide a gateway to lead traffic through your mesh.
```

> * 🚀 Full support for the "inspect API" on builtin gateway resources. This enables users to see which policies impact which `MeshGatewayRoute`.

Reviewer: This sounds more general and makes it less noisier and clearer to understand for anyone not yet aware of the `MeshGatewayRoute` resource
```suggestion
* 🚀 Full support for the "inspect API" on builtin gateway resources. This enables users to see which policies impact which gateway routes.
```

* 🚀 ZoneEgress received many improvements like: support for Standalone, locality aware routing on external services and support for `FaultInjection` and `RateLimit` policies on external services.

> * 🚀 A preview of the completely rewritten transparent-proxy, this aims to make transparent proxy more stable and provide us with pathways for further innovation.

Reviewer: ```suggestion
* 🚀 A preview of the completely rewritten transparent proxy, this aims to make transparent proxy more stable and provide us with pathways for further innovation.
```

> * Many improvements to the Helm charts like: exposing the CP with an ingress, provide resource limits to components, image tag customization.

Reviewer: ```suggestion
* Many improvements to the Helm charts like: exposing the CP with an ingress, providing resource limits to components and customizing image tags.
```

> * A new metric to see how long configuration change take to propagate to data plane proxies.

Reviewer: ```suggestion
* A new metric to see how long configuration changes take to propagate to data plane proxies.
```

And a lot more! The full changelog is available [here](https://github.com/kumahq/kuma/blob/master/CHANGELOG.md).

## Kubernetes Gateway API (experimental)

> The [gateway API kubernetes sig](https://gateway-api.sigs.k8s.io/) is gaining traction and quickly converging to be a flexible successor for Ingress in Kubernetes.

Reviewer: ```suggestion
The [gateway API kubernetes sig](https://gateway-api.sigs.k8s.io/) is gaining traction and quickly converging to be a flexible successor to Ingress in Kubernetes.
```

Since Kuma 1.5.0 we support a builtin Envoy Gateway fully configurable with Kuma policies.
It made sense for us to make the Kubernetes story complete by also providing Gateway API support.

> Gateway API is currently experimental and is enabled with the kuma-cp flag `--experimental-gatewayapi` and will also need the Gateway to be enabled with `--experimental-meshgateway`.

Reviewer: ```suggestion
Gateway API is currently experimental and is enabled with the `kuma-cp` flag `--experimental-gatewayapi` and will also need the Gateway to be enabled with `--experimental-meshgateway`.
```

Check the [Gateway API docs](https://kuma.io/docs/1.6.x/explore/gateway-api) for more information.

## Transparent-proxy rewrite (experimental)

> We have always been very limited by our current transparent proxy. The code-base is mainly inherited and hard to maintain.

Reviewer: ```suggestion
The current transparent proxy implementation presents both a maintenance burden as well as a limit on our options for innovation.
```

> We have big plans on transparent-proxy including some features using cutting edge network and kernel technologies.

Reviewer: ```suggestion
We have big plans for transparent proxy including some features using cutting edge network and kernel technologies.
```

> As a first step in this effort we've completely rewritten the iptables transparent-proxy rules.

Reviewer: ```suggestion
As a first step in this effort we've completely rewritten the iptables transparent proxy rules.
```

> This is now ready and can be enabled with `kumactl install transparent-proxy --experimental-transparent-proxy-engine`.

Reviewer: ```suggestion
We're releasing this rewrite as an experimental feature in version 1.6.0. It can be enabled with `kumactl install transparent-proxy --experimental-transparent-proxy-engine`.
```

> Please take it for a run in dev (this is not ready for production yet) and let us know of any [issue](https://github.com/kumahq/kuma/issues/new)!

Reviewer: ```suggestion
Please take it for a run in dev (this is not ready for production yet) and let us know of any [issues](https://github.com/kumahq/kuma/issues/new)!
```

## ZoneEgress improvements

> In Kuma 1.5.0 ZoneEgress was released, it enables routing all traffic external to the mesh through a dedicated set of proxies.

Reviewer: ```suggestion
In Kuma 1.5.0 zone egress was released, it enables routing all traffic external to the mesh through a dedicated set of proxies.
```

my preference at least. Feels jarring to refer to a whole feature by the resource (and otherwise should be `ZoneEgress`), but may just be me and I don't know what other projects do.

This was a very big step towards our goal to provide best in class support for external services.

In this release we improved ZoneEgress even more with the following features:

You now need to [opt-in your mesh](https://kuma.io/docs/1.6.x/explore/zoneegress/#configuration) to route traffic through the ZoneEgress.

> Previously ZoneEgress routing would be dependent on the presence of a ZoneEgress instance, this would reconfigure data plane proxies if ZoneEgresses where shutdown.

Reviewer: ```suggestion
Previously ZoneEgress routing would be dependent on the presence of a ZoneEgress instance, which would cause data plane proxies to be reconfigured if ZoneEgresses where shutdown.
```

I think

> With this feature routing is not dependent on the state of your ZoneEgress instances.

Reviewer: ```suggestion
With this feature, routing is not dependent on the state of your ZoneEgress instances.
```

When using external service you can now force the traffic to exit your mesh in a specific zone.
This is useful if you want to leverage all the features Kuma provides until you are very close to the service that is outside the mesh.

> To make this happen simply add a `kuma.io/zone` to your external service definition.

Reviewer: ```suggestion
To make this happen simply add a `kuma.io/zone` annotation to your external service definition.
```

or is it label?

Author: it's a tag

Check the [external service docs](https://kuma.io/docs/1.6.x/policies/external-services/#external-services-accessible-from-specific-zone-through-zoneegress) for more details.

> One of the benefit of ZoneEgress is routing all external service traffic through a set of instances, this has the benefit of providing higher control on the network traffic and make external service closer
> to regular data planes.

Reviewer: ```suggestion
One of the benefits of ZoneEgress is that all external service traffic is routed through a set of instances. This has the benefit of providing more control over network traffic and puts external services closer to regular data planes.
```

> In Kuma 1.6.0 you can also do all this in Standalone mode!

Reviewer: ```suggestion
In Kuma 1.6.0 you can do all of this in Standalone mode, too!
```

To learn more about ZoneEgress check out the [docs](https://kuma.io/docs/1.6.x/explore/zoneegress).

## Configuration propagation metric

> When operating a service mesh one thing you need to know is how long it takes for your dataplanes to receive a configuration change.

Reviewer: ```suggestion
When operating a service mesh one thing you need to be aware of is how long it takes for your data plane proxies to receive a configuration change.
```

or even

```suggestion
When operating a service mesh one thing you need to be aware of is how long it takes for your data plane to receive configuration changes.
```

This is now possible with the metric: `xds_delivery`.

For example, we see the impact of [`reachableServices`](https://kuma.io/docs/1.6.x/networking/transparent-proxying/#reachable-services) thanks to this metric.
In the graph below we first deploy 50 services with 3 instances first without `reachableServices` and then do the same thing with `reachableService`.

> Thanks to this new stats you can see that `reachableService` significantly improve configuration delivery time.

Reviewer: ```suggestion
Thanks to this new stat, you can see that `reachableServices` significantly improves configuration delivery time.
```

![XDS_delivery grafana example](/images/blog/kuma1.6.0_xds_delivery.png)

> A chart to track this stat has been added to the control-plane dashboard which helps operate your meshes running on Kuma.

Reviewer: ```suggestion
A chart to track this stat has been added to the control-plane dashboard.
```

not sure, feels unnecessary or should be reworded.

## Kuma 1.5.1

We are also releasing a patch release of Kuma 1.5.x with some critical bug fixes.
Make sure to check the [CHANGELOG](https://github.com/kumahq/kuma/blob/master/CHANGELOG.md).

## Join us on the community call!

> Join us on our [community channels](https://kuma.io/community/) to learn more about Kuma, including our official Slack chat. The community channels are useful to get up and running with Kuma, as well as to learn how to contribute to and discuss the project roadmap. Kuma is a CNCF Sandbox project: neutral, open and inclusive.

Reviewer: ```suggestion
Join us on our [community channels](https://kuma.io/community/), including our official Slack chat, to learn more about Kuma. The community channels are useful for getting up and running with Kuma, as well as for learning how to contribute to and discuss the project roadmap. Kuma is a CNCF Sandbox project: neutral, open and inclusive.
```

> The next community call will be hosted on [March 9th at 8:30am PDT](https://kuma.io/community/). Also don’t forget to follow Kuma [on Twitter](https://twitter.com/kumamesh) and star it [on GitHub](https://github.com/kumahq/kuma)!

Reviewer: ```suggestion
The next community call will be hosted on [March 9th at 8:30am PDT](https://kuma.io/community/). And don’t forget to follow Kuma [on Twitter](https://twitter.com/kumamesh) and star it [on GitHub](https://github.com/kumahq/kuma)!
```

---
title: Kuma 1.6.0 GA Released With Kubernetes Gateway API support, ZoneEgress improvements, a new and improved transparent-proxy and more.
description: Kuma 1.6.0 GA Released With Kubernetes Gateway API support, ZoneEgress improvements, a new and improved transparent-proxy and more.
date: 2022-04-12
headerImage: /images/blog/kuma_1_6_blog_banner.jpg
tags:
  - Release
canonicalUrl: 'https://kuma.io/blog/kuma-1-6-0-released'
---

> We are happy to announce Kuma's latest release, which is packed with features and improvements, including substantial performance improvements to our built gateway and .

Reviewer 2: ".. and .." ?

In the graph below we first deploy 50 services with 3 instances first without `reachableServices` and then do the same thing with `reachableServices`.

Thanks to this new metric, you can see that annotating your instances with `kuma.io/transparent-proxying-reachable-services` significantly improve configuration delivery time.

![XDS_delivery grafana example](/images/blog/kuma1.6.0_xds_delivery.png)

A chart to track this metric has been added to the control-plane dashboard.

## Kuma 1.5.1

We are also releasing a patch release of Kuma 1.5.x with some critical bug fixes.
Make sure to check the [CHANGELOG](https://github.com/kumahq/kuma/blob/master/CHANGELOG.md).

## Join us on the community call!

Join us on our [community channels](https://kuma.io/community/), including our official Slack chat, to learn more about Kuma.
The community channels are useful for getting up and running with Kuma, as well as for learning how to contribute to and discuss the project roadmap.
Kuma is a CNCF Sandbox project: neutral, open and inclusive.

> The next community call will be hosted on [March 9th at 8:30am PDT](https://kuma.io/community/).

Reviewer 2: The date needs to be updated.
